pkg/option/config.go

```diff
@@ -3772,26 +3772,36 @@ func (c *DaemonConfig) checkMapSizeLimits() error {
 }
 
 func (c *DaemonConfig) checkIPv4NativeRoutingCIDR() error {
-	if c.GetIPv4NativeRoutingCIDR() == nil && c.EnableIPv4Masquerade && !c.TunnelingEnabled() &&
-		c.IPAMMode() != ipamOption.IPAMENI && c.EnableIPv4 && c.IPAMMode() != ipamOption.IPAMAlibabaCloud {
+	if c.GetIPv4NativeRoutingCIDR() == nil &&
+		c.EnableIPv4 && c.EnableIPv4Masquerade &&
+		!c.EnableIPMasqAgent &&
+		!c.TunnelingEnabled() &&
+		c.IPAMMode() != ipamOption.IPAMENI && c.IPAMMode() != ipamOption.IPAMAlibabaCloud {
 		return fmt.Errorf(
 			"native routing cidr must be configured with option --%s "+
-				"in combination with --%s --%s=%s --%s=%s --%s=true",
-			IPv4NativeRoutingCIDR, EnableIPv4Masquerade, RoutingMode, RoutingModeNative,
-			IPAM, c.IPAMMode(), EnableIPv4Name)
+				"in combination with --%s=true --%s=true --%s=false --%s=%s --%s=%s",
+			IPv4NativeRoutingCIDR,
+			EnableIPv4Name, EnableIPv4Masquerade,
+			EnableIPMasqAgent,
+			RoutingMode, RoutingModeNative,
+			IPAM, c.IPAMMode())
 	}
 
 	return nil
```

Comment:
Given that this helper is now pretty complicated, I'm wondering if reversing the conditions and using early return might improve readability:

```
	if c.GetIPv4NativeRoutingCIDR() != nil {
		return nil
	}
	if !c.EnableIPv4 || !c.EnableIPv4Masquerade {
		return nil
	}
	if c.EnableIPMasqAgent {
		return nil
	}
	if c.TunnelingEnabled() {
		return nil
	}
	if c.IPAMMode() == ipamOption.IPAMENI || c.IPAMMode() == ipamOption.IPAMAlibabaCloud {
		return nil
	}

	return fmt.Errorf(
		"native routing cidr must be configured with option --%s "+
			"in combination with --%s=true --%s=true --%s=false --%s=%s --%s=%s",
		IPv4NativeRoutingCIDR,
		EnableIPv4Name, EnableIPv4Masquerade,
		EnableIPMasqAgent,
		RoutingMode, RoutingModeNative,
		IPAM, c.IPAMMode())
```

Reply:
Yeah good point! I'll change it